AcadeMeData/models.py

```diff
@@ -199,3 +199,52 @@ def __str__(self):
 
     def get_msg_tag(id):
         return MessageTags.objects.get(id=id)
+
+
+class Course(models.Model):
+    course_id = models.IntegerField(primary_key=True, validators=[MinValueValidator(0)], default=0)
+    name = models.CharField(max_length=100)  # Name of the course
+    degree = models.ForeignKey(Degree, on_delete=models.RESTRICT)
+    elective = models.BooleanField(default=False)  # False for mandatory, True for elective
+    description = models.TextField(null=True, blank=True)  # Decribes the course
+    professor = models.ForeignKey(Professor, on_delete=models.RESTRICT)
+
+    # Fields for a rating system we will implement later:
+    # The rating of this course.
+    rating = models.DecimalField(max_digits=2, decimal_places=1,
+                                 validators=[MinValueValidator(1), MaxValueValidator(5)], blank=True, null=True)
+    # Counter for the number of rates, will be used later to calculate the rating.
+    sum_rates = models.IntegerField(
+        validators=[MinValueValidator(0)], default=0)
+
+    # methods
+    def __str__(self):
+        """
+        Returns the name of all possible courses in the database.
+        """
+        return self.course
```

Comment:
not tested. did you check it works?

Reply:
As we have no use for this function, I've decided to remove it.